pdfbox/__init__.py

```diff
@@ -176,3 +176,73 @@ def extract_text(self, input_path, output_path='',
         p = sarge.capture_stdout(cmd)
         if not output_path:
             return p.stdout.text
+
+    def pdf_to_images(self, input_path, output_path, password=None,
+                      imageType=None, outputPrefix=None,
+                      startPage=None, endPage=None,
+                      page=None, dpi=None, color=None, cropbox=None,time=None):
+        """
+        Extract all text from PDF file.
+
+        Parameters
+        ----------
+        input_path : str
+            Input PDF file.
+        output_path : str
+            Output text file. If not specified, the extracted text is returned.
+        password : str
+            PDF password.
+        imageType : str
+            The image type to write to. Currently only jpg or png.
+            Default: jpg.
+        outputPrefix : str
+            The prefix to the image file.
+            DEfault: Name of PDF document.
+        startPage : bool
+            The first page to convert, one based.
+            Default: 1.
+        endPage : bool
+            The last page to convert, one based.
+            Default: Last Page.
+        page : int
+            The only page to extract (1-based).
+        dpi : int
+            DPI resolution of exported images.
+            Default: Detected from screen (or 96 if headless).
+        color : int
+            The color depth (valid: bilevel, gray, rgb, rgba).
+            Default: rgb.
+        cropbox : str
+            The page area to export.
+            Format: e.g "34 45 56 67"
+        time : int
+            Prints timing information to stdout.
+
+        Returns
+        -------
+        text : str
+            Time taken to complete the process.
+        """
+
+        options = (' -password {password}'.format(password=password) if password else '') + \
+                  (' -imageType {imageType}'.format(imageType=imageType) if imageType else '') + \
+                  (' -outputPrefix {outputPrefix}'.format(outputPrefix=outputPrefix) if outputPrefix else '') + \
+                  (' -startPage {startPage}'.format(startPage=startPage) if startPage else '') + \
+                  (' -endPage {endPage}'.format(endPage=endPage) if endPage else '') + \
+                  (' -page {page}'.format(page=page) if page else '') + \
+                  (' -dpi {dpi}'.format(dpi=dpi) if dpi else '') + \
+                  (' -color {color}'.format(color=color) if color else '') + \
+                  (' -cropbox {cropbox}'.format(cropbox=cropbox) if cropbox else '') + \
+                  (' -time {time}'.format(time=time) if time else '')
+
+        dst = os.path.join(output_path,os.path.basename(input_path))
+        dst = os.path.abspath(dst)
+        shutil.copy(input_path,dst)
```

Comment:
Why is it necessary to copy the input file to the output directory? Doing so is dangerous because it silently will overwrite any file with the same name in that directory without the user's knowledge.

Reply:
The command can only create the Images to the location where the input file is. In case the user wants to have the output file at different location the file must be present there. I find this to be one of the workaround. Any suggestions ?

Reply:
Since there already is an `outputPrefix` option that can be used to direct the output to a different directory than the input, I recommend removing the `output_path` option altogether so that the Python interface is as close to that of the underlying Java as possible.